modules/ximgproc/src/deriche_filter.cpp

```diff
@@ -0,0 +1,410 @@
+#include "precomp.hpp"
+#include "opencv2/highgui.hpp"
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+namespace cv {
+namespace ximgproc {
+    template<typename T>
+    static void VerticalIIRFilter(Mat img,Mat dst,const Range r,T *c1,double alphaDerive)
+    {
+        float                *f2;
+        int tailleSequence = (img.rows>img.cols) ? img.rows : img.cols;
+        Mat matG1(1, tailleSequence, CV_64FC1), matG2(1, tailleSequence, CV_64FC1);
+        double *g1 = matG1.ptr<double>(0), *g2 = (double*)matG2.ptr<double>(0);
+        double    kp = pow(1 - exp(-alphaDerive), 2.0) / exp(-alphaDerive);
+        double a1, a2, a3, a4;
+        double b1, b2;
+        int rows = img.rows, cols = img.cols;
+
+        kp = pow(1 - exp(-alphaDerive), 2.0) / exp(-alphaDerive);
+        a1 = 0;
+        a2 = kp*exp(-alphaDerive), a3 = -kp*exp(-alphaDerive);
+        a4 = 0;
+        b1 = 2 * exp(-alphaDerive);
+        b2 = -exp(-2 * alphaDerive);
+        for (int j = r.start; j<r.end; j++)
+        {
+            // Causal vertical  IIR filter
+            c1 = img.ptr<T>(0);
+            f2 = dst.ptr<float>(0);
+            f2 += j;
+            c1 += j;
+            int i = 0;
+            g1[i] = (a1 + a2)* *c1;
+            i++;
+            c1 += cols;
```

Comment:
`c1 += cols;`
This works for contiguous matrix only.

Probably you need this: `T value = img.at<T>(row, col);`